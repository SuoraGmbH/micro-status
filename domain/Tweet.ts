export interface Tweet {
  userId: string;
  message: string;
  date: string;
  id: number;
}
