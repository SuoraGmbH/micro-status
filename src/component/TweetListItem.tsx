import * as React from 'react';
import { Tweet } from '../domain/Tweet';

interface Props {
  tweet: Tweet;
}

export class TweetListItem extends React.Component<Props> {
  public render() {
    const { tweet } = this.props;
    return (
      <div>
        <h6>{tweet.userId}</h6>
        <p>{tweet.message}</p>
      </div>
    );
  }
}
