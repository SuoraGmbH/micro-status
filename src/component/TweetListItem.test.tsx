import * as React from 'react';
import * as ReactDOM from 'react-dom';
import * as renderer from 'react-test-renderer';
import { Tweet } from '../domain/Tweet';
import { TweetListItem } from './TweetListItem';

const tweet: Tweet = {
  id: 1,
  date: '2018-09-23T12:25:41.801Z',
  message: 'Hello World!',
  userId: 'muhdiekuh',
};

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<TweetListItem tweet={tweet} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders and has a child', () => {
  const div = document.createElement('div');
  ReactDOM.render(<TweetListItem tweet={tweet} />, div);

  const header = div.childNodes[0].childNodes[1].childNodes[0].childNodes[0];

  expect(header.nodeName).toEqual('H6');
  expect(header.textContent).toEqual('muhdiekuh');

  ReactDOM.unmountComponentAtNode(div);
});

it('renders according to snapshot', () => {
  const tree = renderer.create(<TweetListItem tweet={tweet} />).toJSON();
  expect(tree).toMatchSnapshot();
});
