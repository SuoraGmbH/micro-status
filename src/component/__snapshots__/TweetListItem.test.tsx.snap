// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders according to snapshot 1`] = `
<li>
  <div
    className="avatar"
  >
    <img
      src="http://i.pravatar.cc/100?u=muhdiekuh"
    />
  </div>
  <div
    className="tweet-wrapper"
  >
    <div
      className="tweet"
    >
      <h6>
        muhdiekuh
      </h6>
      <p>
        Hello World!
      </p>
    </div>
    <div
      className="arrow"
    />
  </div>
</li>
`;
